firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift

```diff
@@ -437,10 +438,14 @@ class ToolbarMiddleware: FeatureFlaggable {
 
         let numberOfTabs = action.numberOfTabs ?? toolbarState.numberOfTabs
         let isShowMenuWarningAction = action.actionType as? ToolbarMiddlewareActionType == .showMenuWarningBadge
-        let menuBadgeImageName = isShowMenuWarningAction ? action.badgeImageName : toolbarState.menuWarningBadge
+        let (menuBadgeImageName, maskImageName) = if isShowMenuWarningAction {
+                                                    (action.badgeImageName, action.maskImageName)
+                                                } else {
+                                                    (toolbarState.badgeImageName, toolbarState.maskImageName)
+                                                }
```

Comment:
Just a nit because I don't like tuples and because this code is shorter:
```suggestion
        let menuBadgeImageName = isShowMenuWarningAction ? action.badgeImageName : toolbarState.badgeImageName
        let maskImageName = isShowMenuWarningAction ? action.maskImageName : toolbarState.maskImageName
```